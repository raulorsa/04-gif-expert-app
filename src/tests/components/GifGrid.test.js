
import React from 'react';

import { shallow } from "enzyme";
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { GifGrid } = require("../../components/GifGrid");
jest.mock('../../hooks/useFetchGifs.js');

describe('Priebas con el GifGrid', () =>{

    const category = 'Dragon Ball';

    
    
    test('Debe mostrarse correctamente', () => {
        
        const gifs =[{
            id: 'ABC',
            url: 'https://localhost/nada/hola.jpg',
            title: 'prueba de verdad'
        }]

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={ category} />);
        expect (wrapper).toMatchSnapshot();
    });
    
    test('debe de mostrat items cuando se carga imagenes', () => {
        
        
        const gifs =[{
            id: 'ABC',
            url: 'https://localhost/nada/hola.jpg',
            title: 'prueba de verdad'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={ category} />);
        expect (wrapper).toMatchSnapshot();

    })


})