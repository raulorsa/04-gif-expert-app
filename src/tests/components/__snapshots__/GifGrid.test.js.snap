// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Priebas con el GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
     
    Dragon Ball
     
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeInDown"
  />
</Fragment>
`;

exports[`Priebas con el GifGrid debe de mostrat items cuando se carga imagenes 1`] = `
<Fragment>
  <h3>
     
    Dragon Ball
     
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeInDown"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="prueba de verdad"
      url="https://localhost/nada/hola.jpg"
    />
  </div>
</Fragment>
`;
